Model/ResourceModel/Currency.php

```diff
@@ -36,8 +36,51 @@ public function getByCurrency($code)
         return $connection->fetchAll($select);
     }
 
+    //should write  deletion routine too
+
+    /**
+     * To retrieve precision of a currency from the database.
+     *
+     * @param $currencyCode string
+     * @return array
+     */
+    public function getPrecisionByCurrency($currencyCode)
+    {
+        $connection = $this->getConnection();
+        $select = $connection->select()
+            ->from($this->getTable('reach_currency_precision'))
+            ->where('currency_code = ?', $currencyCode);
+        return $connection->fetchAssoc($select);
+    }
+
+    /**
+     * To save precision of a currency into the database.
+     *
+     * https://framework.zend.com/manual/1.12/en/zend.db.adapter.html says:
+     * The return value of the insert() method is not the last inserted ID, because the table 
+     * might not have an auto-incremented column. Instead, the return value is the number of 
+     * rows affected (usually 1).
+     * If your table is defined with an auto-incrementing primary key, you can call the lastInsertId()
+     * method after the insert. This method returns the last value generated in the scope of the current
+     * database connection.
+
+     * @param $currencyCode string
+     * @param $precision integer
+     * @return \Zend_Db_Statement_Interface
+     */
+    public function setPrecisionByCurrency($currencyCode, $precision)
+    {
+        $connection = $this->getConnection();
+
+        $connection->insert('reach_currency_precision', [
```

Comment:
would it be worthwhile to handle the case when insert doesn't actually update anything? (if 0 is returned by the insert() vs what we expected (1))

Reply:
discussed with Suraiya, she considered throwing an exception in this case but to avoid breaking the site because of a thrown exception we're going to leave this code as is.

If the precision details can't be added to the database for some reason, it should still be cached. If it can't be cached either then another /localize call will be made. If a thousand /localize calls are hitting Reach then we'd notice right away.

It would be an edge edge case for the database to be inaccessible.